extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -360,6 +360,17 @@ public void handle(Void e) {
                 }
             });
         }
+        // Headers sent on any request, regardless of the response
+        Map<String, String> headers = httpConfiguration.headers;
+        if (!headers.isEmpty()) {
+            httpRouteRouter.route().order(-1).handler(new Handler<RoutingContext>() {
```

Comment:
Authentication and Authorization run at -100 and -200, so it won't be set if a request is rejected by the security layer.

Reply:
Which is a good thing, right? 😃

Reply:
You said in a previous comment that you wanted these headers set 'regardless of the outcome'?

Reply:
Right, changed to `Integer.MIN_VALUE`